### PR TITLE
Feat/office365 promo

### PR DESCRIPTION
**Pull request by virgotopia**

### CODE REVIEW

File: src/exchange/information/information.controller.js [hunk 1]
```diff
@@ -35,6 +38,11 @@ angular
             this.loadPtrTooltip();
             this.loadPtrv6Tooltip();
             this.loadATooltip();
+
+            this.officeOfferService.getOfficeOfferSubscription()
+                .then((officeTenantServiceName) => {
+                    this.shouldDisplayOfficeAttached = !!officeTenantServiceName;
```
Comment (FredericEspiau) [inline src/exchange/information/information.controller.js]: `_(officeTenantServiceName).isString() && !_(officeTenantServiceName).isEmpty()` would be more explicit

File: src/exchange/information/office-offer/office-offer.controller.js [hunk 2]
```diff
@@ -192,45 +148,45 @@ angular
...
-                const alreadyPresentAccounts = this.selectedAccounts.map((account) => account.primaryEmailDisplayName);
-                this.selectedAccounts = this.selectedAccounts.concat(accounts.list.results.filter((account) => !alreadyPresentAccounts.includes(account.primaryEmailDisplayName)));
-            }
-
-            this.services.$scope.accounts = accounts;
-        }
-
-        countNumberOfCheckedAccounts () {
+        countNumberOfCheckedAccounts (item) {
             if (!_.isEmpty(this.accounts)) {
```
Comment (FredericEspiau) [inline src/exchange/information/office-offer/office-offer.controller.js]: `_(this.accounts).isArray() && !_(this.accounts).isEmpty()` I always check the type before using `isEmpty` as `isEmpty` would return `true` for numbers.

File: src/exchange/information/office-offer/office-offer.controller.js [hunk 3]
```diff
@@ -192,45 +148,45 @@ angular
...
-                const currentlyDislayedAccountsThatAreSelected = this.accounts.list.results.filter((account) => currentlySelectedAccountsEmailAddresses.includes(account.primaryEmailDisplayName));
-
-                const alreadyPresentAccounts = currentlyDislayedAccountsThatAreSelected.map((account) => account.primaryEmailDisplayName);
-                this.selectedAccounts = currentlyDislayedAccountsThatAreSelected.concat(selectedAccountsCurrentBeingDisplayed.filter((account) => !alreadyPresentAccounts.includes(account.primaryEmailDisplayName)));
-
-                const currentlyNotSelectedAccountsEmailAddresses = Object.keys(this.selectedCheckboxes).filter((key) => !this.selectedCheckboxes[key]);
-
-                this.selectedAccounts = this.selectedAccounts.filter((account) => !currentlyNotSelectedAccountsEmailAddresses.includes(account.primaryEmailDisplayName));
+                const selectedAccountsEmails = this.loadSelectedAccounts();
+                if (item) {
```
Comment (FredericEspiau) [inline src/exchange/information/office-offer/office-offer.controller.js]: `_(item).isObject()` would be more explicit

File: src/css/exchange/exchangeDiagnostic.css [hunk 4]
```diff
@@ -51,25 +51,29 @@
...
+    opacity: 0.3;
 }
 
-.image-container:hover .overlay-transition{
-  opacity: 1;
+.image-container:hover .overlay-transition {
+    opacity: 1;
 }
+
+.ovh-font.ovh-font-0x4 {
```
Comment (FredericEspiau) [inline src/css/exchange/exchangeDiagnostic.css]: This shouldn't be in this file, you should look for a place in the `ovh-manager-web` project or create a stylesheet for office-attach

File: src/exchange/information/information.html [hunk 5]
```diff
@@ -105,6 +104,18 @@
...
                                     </div>
                                 </li>
+                                <li class="oui-tile__item"
+                                    data-ng-if="ctrl.shouldDisplayOfficeAttached">
+                                    <div class="oui-tile__definition panel-body">
+                                        <button class="btn btn-block btn-default ml-auto my-auto" type="button"
+                                                data-ng-click="ctrl.displayOfficeOffer()">
+                                            <strong class="d-block" data-i18n-static="exchange_office_1M_promotion_button_title"></strong>
+                                            <div class="text-right ml-auto my-auto">
+                                                <span class="bottom ovh-font ovh-font-0x8" data-i18n-static="exchange_office_1M_promotion_button_title_extra"></span>
```
Comment (FredericEspiau) [inline src/exchange/information/information.html]: `ovh-font-0x8` is not defined, are you sure it exists ?

File: src/resources/i18n/exchange/Messages_fr_FR.xml [hunk 6]
```diff
@@ -61,24 +61,27 @@
...
-   <translation id="exchange_office_attached_alert_modale_search_form" qtlid="404291">Rechercher des comptes emails à utiliser</translation>
-   <translation id="exchange_office_attached_alert_modale_confirmation_header" qtlid="404304">Vous allez activer l'offre Office 365 sur les {0} comptes suivants :</translation>
-   <translation id="exchange_office_attached_alert_modale_confirmation_checkbox" qtlid="404317">En cochant cette case, je confirme vouloir souscrire à l'offre pour les comptes ci-dessus et je confirme avoir compris que l'offre sera reconduite automatiquement si je ne l'annule pas.</translation>
+<!-- OFFICE - One month free promotion -->
+   <translation id="exchange_office_1M_promotion_button_title">Essai gratuit d'un mois Office 365</translation>
+   <translation id="exchange_office_1M_promotion_button_title_extra">Offre promotionnelle</translation>
+   <translation id="exchange_office_1M_promotion_main_title">Activer l'offre promotionnelle Office 365</translation>
+   <translation id="exchange_office_1M_promotion_intro">Obtenez un mois gratuit d'Office 365  sur un maximum de {0} comptes Exchange.</translation>
```
Comment (FredericEspiau) [inline src/resources/i18n/exchange/Messages_fr_FR.xml]: `{0} compte(s) Exchange`

File: src/resources/i18n/exchange/Messages_fr_FR.xml [hunk 7]
```diff
@@ -61,24 +61,27 @@
...
+   <translation id="exchange_office_1M_promotion_intro_table">Sélectionnez les comptes emails de votre choix</translation>
```
Comment (FredericEspiau) [inline src/resources/i18n/exchange/Messages_fr_FR.xml]: `e-mails`

File: src/resources/i18n/exchange/Messages_fr_FR.xml [hunk 8]
```diff
@@ -61,24 +61,27 @@
...
+   <translation id="exchange_office_1M_promotion_selected_accounts">{0} comptes sélectionnés pour un maximum de {1}</translation>
```
Comment (FredericEspiau) [inline src/resources/i18n/exchange/Messages_fr_FR.xml]: `{0} compte(s) sélectionné(s)`

File: src/resources/i18n/exchange/Messages_fr_FR.xml [hunk 9]
```diff
@@ -61,24 +61,27 @@
...
+   <translation id="exchange_office_1M_promotion_search_form">Rechercher des comptes emails à utiliser</translation>
```
Comment (FredericEspiau) [inline src/resources/i18n/exchange/Messages_fr_FR.xml]: `des comptes e-mails`

File: src/resources/i18n/exchange/Messages_fr_FR.xml [hunk 10]
```diff
@@ -61,24 +61,27 @@
...
+   <translation id="exchange_office_1M_promotion_account_types">Types de comptes à utiliser dans les recherches</translation>
+   <translation id="exchange_office_1M_promotion_confirmation_table">Confirmez les comptes e-mail</translation>
```
Comment (FredericEspiau) [inline src/resources/i18n/exchange/Messages_fr_FR.xml]: `les comptes e-mails`

File: src/resources/i18n/exchange/Messages_fr_FR.xml [hunk 11]
```diff
@@ -61,24 +61,27 @@
...
+   <translation id="exchange_office_1M_promotion_confirmation_header"> {0} sur {1} comptes e-mail sélectionnés</translation>
```
Comment (FredericEspiau) [inline src/resources/i18n/exchange/Messages_fr_FR.xml]: `{0} sur {1} comptes e-mails sélectionnés` (I changed the accent)

File: src/resources/i18n/exchange/Messages_fr_FR.xml [hunk 12]
```diff
@@ -61,24 +61,27 @@
...
+   <translation id="exchange_office_1M_promotion_selection_button_caption">Sélectionner</translation>
+   <translation id="exchange_office_1M_promotion_selection_change_link">Modifier mes choix</translation>
+   <translation id="exchange_office_1M_promotion_legal_conditions_caption">En cochant cette case, je confirme vouloir souscrire à l'offre pour les comptes ci-dessus et je confirme avoir compris que l'offre sera reconduite automatiquement si je ne l'annule pas. (Vous aurez un rappel X jours avant la fin de l'offre promotionnelle.)</translation>
```
Comment (FredericEspiau) [inline src/resources/i18n/exchange/Messages_fr_FR.xml]: Were these messages confirmed by Content ?

Comment (virgotopia) [reply src/resources/i18n/exchange/Messages_fr_FR.xml]: In progress, it takes time...

File: src/exchange/information/office-offer/office-offer.controller.js (hunk 2 above)
Comment (virgotopia) [reply src/exchange/information/office-offer/office-offer.controller.js]: It's not an array but an object. Eventually I can write it like this:
_(this.accounts).isObject()  && !_(this.accounts).isEmpty()

File: src/exchange/information/information.controller.js (hunk 1 above)
Comment (virgotopia) [reply src/exchange/information/information.controller.js]: Wrong parameter name here, it's a boolean coming down the pipe, not a String.
Fixed by renaming .

File: src/exchange/information/information.html (hunk 5 above)
Comment (virgotopia) [reply src/exchange/information/information.html]: Done, it's defined now.

**Comment by antleblanc** [comment]
@virgotopia You should target `develop` instead of `master` branch. @jleveugle Am I right?

Furthermore there are some conflict :candy: :)

**Comment by jleveugle** [comment]
@FredericEspiau 

I close this PR due to inactivity and conflicts.

Feel free to re-open it

**Pull request closed without merging**